optimum/intel/openvino/quantization.py

```diff
@@ -330,25 +328,90 @@ def _quantize_torchmodel(
             model_type=model_type,
         )
 
-        if self.task == "text-generation":
-            onnx_config = onnx_config_class(self.model.config, use_past=self.model.config.use_cache)
+        if task == "text-generation":
+            onnx_config = onnx_config_class(model.config, use_past=model.config.use_cache)
         else:
-            onnx_config = onnx_config_class(self.model.config)
+            onnx_config = onnx_config_class(model.config)
 
-        compressed_model.eval()
-        num_parameters = compressed_model.num_parameters()
-        save_as_external_data = use_external_data_format(num_parameters) or quantization_config.save_onnx_model
-        f = io.BytesIO() if not save_as_external_data else save_directory / ONNX_WEIGHTS_NAME
+        onnx_path = save_directory / ONNX_WEIGHTS_NAME
 
-        # Export the compressed model to the ONNX format
+        # Export the model to the ONNX format
         opset = min(onnx_config.DEFAULT_ONNX_OPSET, MAX_ONNX_OPSET)
-        opset = opset if not quantization_config.save_onnx_model else max(opset, MIN_ONNX_QDQ_OPSET)
-        _onnx_export_nncf_model(compressed_model, onnx_config, f, opset)
+        opset = max(opset, MIN_ONNX_QDQ_OPSET)
+        export(
+            model=compressed_model,
+            config=onnx_config,
+            opset=opset,
+            output=onnx_path,
+        )
 
         # Load and save the compressed model
-        model = core.read_model(f) if save_as_external_data else core.read_model(f.getvalue(), b"")
+        model = core.read_model(onnx_path)
         self._save_pretrained(model, output_path)
         quantization_config.save_pretrained(save_directory)
+        if not quantization_config.save_onnx_model:
+            os.remove(onnx_path)
+            try:
+                os.remove(f"{onnx_path}_data")
+            except FileNotFoundError:
+                pass
+
+    def compress_weights(
+        self, save_directory: Union[str, Path], quantization_config: OVConfig = None, file_name: Optional[str] = None
+    ):
+        """
+        Compress weights to integer precision (8-bit by default) while keeping activations
+        floating-point. Fits best for LLM footprint reduction and performance acceleration.
+
+        Args:
+            save_directory (`Union[str, Path]`):
+                The directory where the quantized model should be saved.
+            quantization_config (`OVConfig`, *optional*):
+                The configuration containing the parameters related to quantization.
+            file_name (`str`, *optional*):
+                The model file name to use when saving the model. Overwrites the default file name `"model.onnx"`.
+
+        Example:
+        ```python
+        >>> from optimum.intel.openvino import OVQuantizer, OVModelForCausalLM
+        >>> from transformers import AutoModelForSequenceClassification
+        >>> model = OVModelForCausalLM.from_pretrained("databricks/dolly-v2-3b")
```

Comment:
Shouldn't it be :
```suggestion
        >>> model = AutoModelForCausalLM.from_pretrained("databricks/dolly-v2-3b")
```
as `_quantize_torchmodel` is called in `compress_weights` ?

Reply:
thanks!